exp/cloud/services/scalesets/vmss.go

```diff
@@ -0,0 +1,200 @@
+
+	vmss := compute.VirtualMachineScaleSet{
+		Location: to.StringPtr(s.Scope.Location()),
+		Tags: converters.TagsToMap(infrav1.Build(infrav1.BuildParams{
+			ClusterName: s.Scope.Name(),
+			Lifecycle:   infrav1.ResourceLifecycleOwned,
+			Name:        to.StringPtr(s.MachinePoolScope.Name()),
+			Role:        to.StringPtr(infrav1.Node),
+			Additional:  additionalTags,
+		})),
+		Sku: &compute.Sku{
+			Name:     to.StringPtr(vmssSpec.Sku),
+			Tier:     to.StringPtr("Standard"),
+			Capacity: to.Int64Ptr(vmssSpec.Capacity),
+		},
+		VirtualMachineScaleSetProperties: &compute.VirtualMachineScaleSetProperties{
+			UpgradePolicy: &compute.UpgradePolicy{
+				Mode: compute.Manual,
+			},
+			VirtualMachineProfile: &compute.VirtualMachineScaleSetVMProfile{
+				OsProfile: &compute.VirtualMachineScaleSetOSProfile{
+					ComputerNamePrefix: to.StringPtr(vmssSpec.Name),
+					AdminUsername:      to.StringPtr(azure.DefaultUserName),
+					CustomData:         to.StringPtr(vmssSpec.CustomData),
+					LinuxConfiguration: &compute.LinuxConfiguration{
+						SSH: &compute.SSHConfiguration{
+							PublicKeys: &[]compute.SSHPublicKey{
+								{
+									Path:    to.StringPtr(fmt.Sprintf("/home/%s/.ssh/authorized_keys", azure.DefaultUserName)),
+									KeyData: to.StringPtr(vmssSpec.SSHKeyData),
+								},
+							},
+						},
+						DisablePasswordAuthentication: to.BoolPtr(true),
+					},
+				},
+				StorageProfile: storageProfile,
+				NetworkProfile: &compute.VirtualMachineScaleSetNetworkProfile{
+					NetworkInterfaceConfigurations: &[]compute.VirtualMachineScaleSetNetworkConfiguration{
+						{
+							Name: to.StringPtr(vmssSpec.Name + "-netconfig"),
+							VirtualMachineScaleSetNetworkConfigurationProperties: &compute.VirtualMachineScaleSetNetworkConfigurationProperties{
+								Primary:            to.BoolPtr(true),
+								EnableIPForwarding: to.BoolPtr(true),
+								IPConfigurations: &[]compute.VirtualMachineScaleSetIPConfiguration{
+									{
+										Name: to.StringPtr(vmssSpec.Name + "-ipconfig"),
+										VirtualMachineScaleSetIPConfigurationProperties: &compute.VirtualMachineScaleSetIPConfigurationProperties{
+											Subnet: &compute.APIEntityReference{
+												ID: to.StringPtr(s.Scope.AzureCluster.Spec.NetworkSpec.Subnets[0].ID),
+											},
+											Primary:                 to.BoolPtr(true),
+											PrivateIPAddressVersion: compute.IPv4,
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	err = s.Client.CreateOrUpdate(
+		ctx,
+		s.Scope.ResourceGroup(),
+		vmssSpec.Name,
+		vmss)
+	if err != nil {
+		return errors.Wrapf(err, "cannot create vmss")
+	}
+
+	klog.V(2).Infof("successfully created vmss %s ", vmssSpec.Name)
+	return nil
+}
+
+func (s *Service) Delete(ctx context.Context, spec interface{}) error {
+	vmSpec, ok := spec.(*Spec)
+	if !ok {
+		return errors.New("invalid vmss specification")
+	}
+	klog.V(2).Infof("deleting vmss %s ", vmSpec.Name)
```

Comment:
nit: I'd prefer using `VMSS` instead of `vmss` everywhere in logs and code comments. I made the equivalent changes to VMs in #544